app/models/gem_content.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class GemContent
+  CHECKSUMS_ROOT_FORMAT = "gems/%{gem}/checksums/"
+  CHECKSUMS_KEY_FORMAT = "gems/%{gem}/checksums/%{version}.sha256"
+  CONTENT_ROOT_FORMAT = "gems/%{gem}/contents/"
+  CONTENT_KEY_FORMAT = "gems/%{gem}/contents/%{hex}"
+  PATH_ROOT_FORMAT = "gems/%{gem}/paths/%{version}/"
+  PATH_KEY_FORMAT = "gems/%{gem}/paths/%{version}/%{path}"
+  SPEC_KEY_FORMAT = "gems/%{gem}/specs/%{gem}-%{version}.gemspec"
+
+  attr_reader :gem
+
+  def initialize(gem:)
+    raise ArgumentError, "gem must be Rubygem#name" unless gem.try(:match?, Rubygem::NAME_PATTERN)
+    @gem = gem
+  end
+
+  def fs
+    RubygemFs.content
+  end
+
+  def get(hex)
+    return if hex.blank?
+    fs.get key(hex)
+  end
+
+  def keys
+    fs.each_key(prefix: root).map { |key| key.delete_prefix root }
+  end
+
+  def store(entry)
+    return unless entry.body_persisted?
+    # response = fs.head content_key(entry.hex)
```

Comment:
commented code

Reply:
haha, I swear I removed this. I'll try again.